### PR TITLE
Run static type checking with Mypy

### LINKED ISSUE #114: Publish with `py.typed`
https://peps.python.org/pep-0561/

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #114

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
`py.typed` is included in the distribution: https://github.com/pytest-dev/pytest-subtests/actions/runs/6951118504?pr=115#summary-18912531061

File: tox.ini
Comment (RonnyPfannschmidt) [inline tox.ini:17]:
```suggestion
deps = 
  mypy
  pytest
  
```

File: pyproject.toml
Comment (RonnyPfannschmidt) [inline pyproject.toml:8]: to be properly usable for externals we need to be closer to strict than the defaults

i haven't yet take an look how much pain is in that change but i believe we need to disallow untyped defs

**Review by RonnyPfannschmidt — CHANGES_REQUESTED** [review]
thanks for getting this started!

File: tox.ini
Comment (edgarrmondragon) [reply tox.ini]: 91b5f1a

File: pyproject.toml
Comment (edgarrmondragon) [reply pyproject.toml:8]: 91b5f1ad40d0d3e4d818ae9377c608cb766cbdbd

**Comment by edgarrmondragon** [comment]
Thanks for the feedback @RonnyPfannschmidt! Please take a look at your convenience.

File: setup.cfg
Comment (nicoddemus) [inline setup.cfg:32]:
```suggestion
    attrs>=19.2.0
    pytest>=7.0
    typing_extensions;python_version<"3.8"
```

**Review by nicoddemus — APPROVED** [review]
Great work, thanks!

**Comment by nicoddemus** [comment]
(Working on the mypy failures)

File: .pre-commit-config.yaml
Comment (nicoddemus) [inline .pre-commit-config.yaml:18]: FYI we in general rely on running mypy with pre-commit instead of having a dedicated job on CI. 👍

**Comment by nicoddemus** [comment]
Thanks again @edgarrmondragon! 

Wait for @RonnyPfannschmidt's review before merging. 👍

**Review by RonnyPfannschmidt — APPROVED** [review]

**Pull request merged**
